### ISSUE TITLE
Crunchyroll broken as of June 2015

### ISSUE THREAD

**Issue by JamesTheAwesomeDude** [rg3/youtube-dl#5881]
Crunchyroll's tightened its security recently..

````
$ youtube-dl http://www.crunchyroll.com/no-game-no-life/episode-12-rule-number-10-653185 -v  
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'http://www.crunchyroll.com/no-game-no-life/episode-12-rule-number-10-653185', u'-v']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.05.20
[debug] Python version 2.7.6 - Linux-3.16.0-34-generic-x86_64-with-LinuxMint-17.1-rebecca
[debug] exe versions: ffmpeg 2.6.3, ffprobe 2.6.3, rtmpdump 2.4
[debug] Proxy map: {}
[Crunchyroll] 653185: Downloading webpage
[Crunchyroll] 653185: Downloading media info
[Crunchyroll] 653185: Downloading media info for 360p
[Crunchyroll] 653185: Downloading media info for 480p
[Crunchyroll] 653185: Downloading media info for 720p
[Crunchyroll] 653185: Downloading media info for 720p
[Crunchyroll] 653185: Downloading media info for 1080p
[Crunchyroll] 653185: Downloading media info for 1080p
[debug] Invoking downloader on 'rtmpe://c18fs.fplive.net/c18?nvb=20150602151523&nva=20150602191623&token=024c09df61cdf17182810'
[download] Destination: No Game No Life Episode 12 – Rule Number 10-653185.flv
[debug] rtmpdump command line: rtmpdump --verbose -r 'rtmpe://c18fs.fplive.net/c18?nvb=20150602151523&nva=20150602191623&token=024c09df61cdf17182810' -o 'No Game No Life Episode 12 – Rule Number 10-653185.flv.part' --playpath mp4:s/ve1433367/video.mp4 --resume --skip 1
[rtmpdump] RTMPDump v2.4
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 2
[rtmpdump] DEBUG: Parsed host    : c18fs.fplive.net
[rtmpdump] DEBUG: Parsed app     : c18?nvb=20150602151523&nva=20150602191623&token=024c09df61cdf17182810
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMPE
[rtmpdump] DEBUG: Hostname : c18fs.fplive.net
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : mp4:s/ve1433367/video.mp4
[rtmpdump] DEBUG: tcUrl    : rtmpe://c18fs.fplive.net:1935/c18?nvb=20150602151523&nva=20150602191623&token=024c09df61cdf17182810
[rtmpdump] DEBUG: app      : c18?nvb=20150602151523&nva=20150602191623&token=024c09df61cdf17182810
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.
[rtmpdump] 0 bytes
[rtmpdump] RTMPDump v2.4
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 2
[rtmpdump] DEBUG: Parsed host    : c18fs.fplive.net
[rtmpdump] DEBUG: Parsed app     : c18?nvb=20150602151523&nva=20150602191623&token=024c09df61cdf17182810
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMPE
[rtmpdump] DEBUG: Hostname : c18fs.fplive.net
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : mp4:s/ve1433367/video.mp4
[rtmpdump] DEBUG: tcUrl    : rtmpe://c18fs.fplive.net:1935/c18?nvb=20150602151523&nva=20150602191623&token=024c09df61cdf17182810
[rtmpdump] DEBUG: app      : c18?nvb=20150602151523&nva=20150602191623&token=024c09df61cdf17182810
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.


ERROR: rtmpdump exited with code 1
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/james/.local/bin/youtube-dl/__main__.py", line 19, in <module>
    youtube_dl.main()
  File "/home/james/.local/bin/youtube-dl/youtube_dl/__init__.py", line 405, in main
    _real_main(argv)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/__init__.py", line 395, in _real_main
    retcode = ydl.download(all_urls)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1486, in download
    res = self.extract_info(url)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 660, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 706, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1157, in process_video_result
    self.process_info(new_info)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1419, in process_info
    success = dl(filename, info_dict)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1361, in dl
    return fd.download(name, info)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/downloader/common.py", line 342, in download
    return self.real_download(filename, info_dict)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/downloader/rtmp.py", line 202, in real_download
    self.report_error('rtmpdump exited with code %d' % retval)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/downloader/common.py", line 155, in report_error
    self.ydl.report_error(*args, **kargs)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 527, in report_error
    self.trouble(error_message, tb)
  File "/home/james/.local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 489, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())
````

**Comment by SirCmpwn** [comment]
:+1:

**Comment by majorcyto** [comment]
They closed my duplicate report.. and yet no one has even looked at this one. O.o

**Comment by fb39ca4** [comment]
I suspect I am having this same problem, but rtmpdump is not displaying debug output. How do I get it to display the debug output on Windows?

**Comment by yan12125** [comment]
@fb39ca4 Add the ```-v``` option.

**Comment by fb39ca4** [comment]
Ok, I get the same output as #5855, showing the encryption key the first time I try to download a video, and then after that, the output is like the one in this issue.

**Comment by dstftw** [comment]
This issue will be fixed in the next version. Thanks for the report.

**Comment by fb39ca4** [comment]
I just tried again using the latest commit, and it is still not working:

    [____@Arch-VM youtube-dl]$ python -m youtube_dl -v -f 480p http://www.crunchyroll.com/fatestay-night/episode-0-prologue-661413
    [debug] System config: []
    [debug] User config: []
    [debug] Command-line args: ['-v', '-f', '480p', 'http://www.crunchyroll.com/fatestay-night/episode-0-prologue-661413']
    [debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
    [debug] youtube-dl version 2015.07.04
    [debug] Git HEAD: ede2144
    [debug] Python version 3.4.3 - Linux-4.0.7-2-ARCH-x86_64-with-arch
    [debug] exe versions: ffmpeg 2.7.1, ffprobe 2.7.1, rtmpdump 2.4
    [debug] Proxy map: {}
    [Crunchyroll] 661413: Downloading webpage
    [Crunchyroll] 661413: Downloading media info
    [Crunchyroll] 661413: Downloading media info for 360p
    [Crunchyroll] 661413: Downloading media info for 480p
    [Crunchyroll] 661413: Downloading media info for 720p
    [Crunchyroll] 661413: Downloading media info for 1080p
    [debug] Invoking downloader on 'rtmpe://c18fs.fplive.net/c18?nvb=20150705012433&nva=20150705052533&token=054ed4f614cb83569dd0d'
    [download] Destination: Fate_stay night – Prologue-661413.flv
    [debug] rtmpdump command line: rtmpdump --verbose -r 'rtmpe://c18fs.fplive.net/c18?nvb=20150705012433&nva=20150705052533&token=054ed4f614cb83569dd0d' -o 'Fate_stay night – Prologue-661413.flv.part' --playpath mp4:s/ve1540635/video.mp4 --resume --skip 1
    [rtmpdump] RTMPDump v2.4
    [rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
    [rtmpdump] DEBUG: Parsing...
    [rtmpdump] DEBUG: Parsed protocol: 2
    [rtmpdump] DEBUG: Parsed host    : c18fs.fplive.net
    [rtmpdump] DEBUG: Parsed app     : c18?nvb=20150705012433&nva=20150705052533&token=054ed4f614cb83569dd0d
    [rtmpdump] DEBUG: Number of skipped key frames for resume: 1
    [rtmpdump] DEBUG: Protocol : RTMPE
    [rtmpdump] DEBUG: Hostname : c18fs.fplive.net
    [rtmpdump] DEBUG: Port     : 1935
    [rtmpdump] DEBUG: Playpath : mp4:s/ve1540635/video.mp4
    [rtmpdump] DEBUG: tcUrl    : rtmpe://c18fs.fplive.net:1935/c18?nvb=20150705012433&nva=20150705052533&token=054ed4f614cb83569dd0d
    [rtmpdump] DEBUG: app      : c18?nvb=20150705012433&nva=20150705052533&token=054ed4f614cb83569dd0d
    [rtmpdump] DEBUG: live     : no
    [rtmpdump] DEBUG: timeout  : 30 sec
    [rtmpdump] DEBUG: Failed to get last keyframe.
    [rtmpdump] DEBUG: Closing connection.
    [rtmpdump] 0 bytes
    [rtmpdump] RTMPDump v2.4
    [rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
    [rtmpdump] DEBUG: Parsing...
    [rtmpdump] DEBUG: Parsed protocol: 2
    [rtmpdump] DEBUG: Parsed host    : c18fs.fplive.net
    [rtmpdump] DEBUG: Parsed app     : c18?nvb=20150705012433&nva=20150705052533&token=054ed4f614cb83569dd0d
    [rtmpdump] DEBUG: Number of skipped key frames for resume: 1
    [rtmpdump] DEBUG: Number of skipped key frames for resume: 1
    [rtmpdump] DEBUG: Protocol : RTMPE
    [rtmpdump] DEBUG: Hostname : c18fs.fplive.net
    [rtmpdump] DEBUG: Port     : 1935
    [rtmpdump] DEBUG: Playpath : mp4:s/ve1540635/video.mp4
    [rtmpdump] DEBUG: tcUrl    : rtmpe://c18fs.fplive.net:1935/c18?nvb=20150705012433&nva=20150705052533&token=054ed4f614cb83569dd0d
    [rtmpdump] DEBUG: app      : c18?nvb=20150705012433&nva=20150705052533&token=054ed4f614cb83569dd0d
    [rtmpdump] DEBUG: live     : no
    [rtmpdump] DEBUG: timeout  : 30 sec
    [rtmpdump] DEBUG: Failed to get last keyframe.
    [rtmpdump] DEBUG: Closing connection.


    ERROR: rtmpdump exited with code 1

**Comment by dstftw** [comment]
This fix works for some locations (e.g. works fo RU, doesn't work for US).

**Comment by JamesTheAwesomeDude** [comment]
Using Python 3 seems to give a slightly more descriptive debugging output:

````
$ python3 -m youtube_dl -v http://www.crunchyroll.com/puella-magi-madoka-magica/episode-12-my-very-best-friend-591757
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-v', 'http://www.crunchyroll.com/puella-magi-madoka-magica/episode-12-my-very-best-friend-591757']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.07.04
[debug] Git HEAD: 89d42c2
[debug] Python version 3.4.0 - Linux-3.16.0-34-generic-x86_64-with-LinuxMint-17.1-rebecca
[debug] exe versions: ffmpeg 2.7.1, ffprobe 2.7.1, rtmpdump 2.4
[debug] Proxy map: {}
[Crunchyroll] 591757: Downloading webpage
[Crunchyroll] 591757: Downloading media info
[Crunchyroll] 591757: Downloading media info for 360p
[Crunchyroll] 591757: Downloading media info for 480p
[Crunchyroll] 591757: Downloading media info for 720p
[Crunchyroll] 591757: Downloading media info for 720p
[Crunchyroll] 591757: Downloading media info for 1080p
[Crunchyroll] 591757: Downloading media info for 1080p
[debug] Invoking downloader on 'rtmpe://c21fs.fplive.net/c21?nvb=20150705213537&nva=20150706013637&token=0eb7e09196c51fe4fa5f6'
[download] Destination: Puella Magi Madoka Magica Episode 12 – My Very Best Friend-591757.flv
[debug] rtmpdump command line: rtmpdump --verbose -r 'rtmpe://c21fs.fplive.net/c21?nvb=20150705213537&nva=20150706013637&token=0eb7e09196c51fe4fa5f6' -o 'Puella Magi Madoka Magica Episode 12 – My Very Best Friend-591757.flv.part' --playpath mp4:s/ve807957/video.mp4 --resume --skip 1
[rtmpdump] RTMPDump v2.4
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 2
[rtmpdump] DEBUG: Parsed host    : c21fs.fplive.net
[rtmpdump] DEBUG: Parsed app     : c21?nvb=20150705213537&nva=20150706013637&token=0eb7e09196c51fe4fa5f6
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMPE
[rtmpdump] DEBUG: Hostname : c21fs.fplive.net
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : mp4:s/ve807957/video.mp4
[rtmpdump] DEBUG: tcUrl    : rtmpe://c21fs.fplive.net:1935/c21?nvb=20150705213537&nva=20150706013637&token=0eb7e09196c51fe4fa5f6
[rtmpdump] DEBUG: app      : c21?nvb=20150705213537&nva=20150706013637&token=0eb7e09196c51fe4fa5f6
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Setting buffer time to: 36000000ms
[rtmpdump] Connecting ...
[rtmpdump] DEBUG: RTMP_Connect1, ... connected, handshaking
[rtmpdump] DEBUG: HandShake: Client type: 06
[rtmpdump] DEBUG: HandShake: DH pubkey position: 472
[rtmpdump] DEBUG: HandShake: Client digest offset: 1383
[rtmpdump] DEBUG: HandShake: Initial client digest: 
[rtmpdump] DEBUG: aa 8d c0 9c 28 0b 9a e4 95 89 ec 3f 10 4d c5 29
[rtmpdump] DEBUG: 8a f9 43 80 1f 99 5f 06 a7 b0 48 39 0d 00 23 43
[rtmpdump] DEBUG: HandShake: Type Answer   : 09
[rtmpdump] WARNING: HandShake: Type mismatch: client sent 6, server answered 9
[rtmpdump] DEBUG: HandShake: Server Uptime : 446068419
[rtmpdump] DEBUG: HandShake: FMS Version   : 4.5.3.1
[rtmpdump] DEBUG: HandShake: Server DH public key offset: 510
[rtmpdump] DEBUG: HandShake: Secret key: 
[rtmpdump] DEBUG: 89 94 40 75 73 4a 40 b5 55 c2 56 f5 39 96 ba 53
[rtmpdump] DEBUG: 9e 17 9a 93 37 dd 42 cf 9e e2 f3 2c 55 7e 01 25
[rtmpdump] DEBUG: ee d7 f0 43 27 49 e6 cc 44 76 36 58 10 f5 20 96
[rtmpdump] DEBUG: ba a4 fa ec e0 f4 3e 9c 18 fc 52 8d 08 f5 12 17
[rtmpdump] DEBUG: f3 56 68 e8 b1 98 0d e9 28 4e 0e 2f 2c 85 52 62
[rtmpdump] DEBUG: 08 60 3d 8b 67 1b d7 02 1f 10 e0 d6 53 ef 63 2f
[rtmpdump] DEBUG: 90 9d d1 c9 42 7d 34 73 0e 24 d3 48 e1 73 e2 21
[rtmpdump] DEBUG: 5c ed 81 c0 2f ea 35 59 f0 35 54 d5 53 04 82 e9
[rtmpdump] DEBUG: RC4 Out Key: 
[rtmpdump] DEBUG: 7a 1e 8b 24 45 a1 8c d9 73 86 82 81 ad 19 1f 0e
[rtmpdump] DEBUG: RC4 In Key: 
[rtmpdump] DEBUG: 78 4c 38 0d 21 76 de c4 dd 26 d1 71 dc 6f 08 3a
[rtmpdump] DEBUG: HandShake: Calculated digest key from secure key and server digest: 
[rtmpdump] DEBUG: 67 c6 bb ae 77 23 4d 9e 8a 5a 17 7b b5 f9 f5 35
[rtmpdump] DEBUG: ba 9f 5f d6 c5 d3 61 04 80 65 da d3 dd 8e ab 58
[rtmpdump] DEBUG: HandShake: Client signature calculated:
[rtmpdump] DEBUG: 5c 66 30 2b 34 9d 7c e1 50 8a ec e7 20 aa 96 a0
[rtmpdump] DEBUG: 76 65 a1 eb 3f b6 14 14 14 aa 55 cc 08 d5 e3 73
[rtmpdump] DEBUG: HandShake: Server sent signature:
[rtmpdump] DEBUG: 08 10 ff b5 3b c4 8b 23 7d f8 09 43 2d f4 04 08
[rtmpdump] DEBUG: 1c 5e b6 dd 7e fa 3d a0 3f 75 b9 44 26 07 a5 17
[rtmpdump] DEBUG: HandShake: Digest key: 
[rtmpdump] DEBUG: 43 31 07 d2 b3 cd 6b 7d df e8 61 7f 92 1d bb 69
[rtmpdump] DEBUG: de 6a fa 76 17 aa fc 4d 1e 4b 29 bc 36 ce 05 5f
[rtmpdump] DEBUG: HandShake: Signature calculated:
[rtmpdump] DEBUG: 08 10 ff b5 3b c4 8b 23 7d f8 09 43 2d f4 04 08
[rtmpdump] DEBUG: 1c 5e b6 dd 7e fa 3d a0 3f 75 b9 44 26 07 a5 17
[rtmpdump] DEBUG: HandShake: Genuine Adobe Flash Media Server
[rtmpdump] DEBUG: HandShake: Handshaking finished....
[rtmpdump] DEBUG: RTMP_Connect1, handshaked
[rtmpdump] DEBUG: Invoking connect
[rtmpdump] INFO: Connected...
[rtmpdump] DEBUG: HandleChangeChunkSize, received: chunk size change to 128
[rtmpdump] DEBUG: RTMP_ClientPacket, received: invoke 165 bytes
[rtmpdump] DEBUG: (object begin)
[rtmpdump] DEBUG: Property: NULL
[rtmpdump] DEBUG: (object begin)
[rtmpdump] DEBUG: Property: <Name:              level, STRING:	error>
[rtmpdump] DEBUG: Property: <Name:               code, STRING:	NetConnection.Connect.Rejected>
[rtmpdump] DEBUG: Property: <Name:        description, STRING:	Connection failed.>
[rtmpdump] DEBUG: Property: <Name:        description, STRING:	[ Server.Reject ] : Connection failed.>
[rtmpdump] DEBUG: (object end)
[rtmpdump] DEBUG: (object end)
[rtmpdump] DEBUG: HandleInvoke, server invoking <_error>
[rtmpdump] ERROR: rtmp server sent error
[rtmpdump] DEBUG: RTMP_ClientPacket, received: invoke 18 bytes
[rtmpdump] DEBUG: (object begin)
[rtmpdump] DEBUG: Property: NULL
[rtmpdump] DEBUG: (object end)
[rtmpdump] DEBUG: HandleInvoke, server invoking <close>
[rtmpdump] ERROR: rtmp server requested close
[rtmpdump] DEBUG: Closing connection.
[rtmpdump] 0 bytes
[rtmpdump] RTMPDump v2.4
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 2
[rtmpdump] DEBUG: Parsed host    : c21fs.fplive.net
[rtmpdump] DEBUG: Parsed app     : c21?nvb=20150705213537&nva=20150706013637&token=0eb7e09196c51fe4fa5f6
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMPE
[rtmpdump] DEBUG: Hostname : c21fs.fplive.net
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : mp4:s/ve807957/video.mp4
[rtmpdump] DEBUG: tcUrl    : rtmpe://c21fs.fplive.net:1935/c21?nvb=20150705213537&nva=20150706013637&token=0eb7e09196c51fe4fa5f6
[rtmpdump] DEBUG: app      : c21?nvb=20150705213537&nva=20150706013637&token=0eb7e09196c51fe4fa5f6
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.


ERROR: rtmpdump exited with code 1
  File "/usr/lib/python3.4/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.4/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/youtube-dl/youtube_dl/__main__.py", line 19, in <module>
    youtube_dl.main()
  File "/tmp/youtube-dl/youtube_dl/__init__.py", line 406, in main
    _real_main(argv)
  File "/tmp/youtube-dl/youtube_dl/__init__.py", line 396, in _real_main
    retcode = ydl.download(all_urls)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 1501, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 665, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 711, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 1171, in process_video_result
    self.process_info(new_info)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 1433, in process_info
    success = dl(filename, info_dict)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 1375, in dl
    return fd.download(name, info)
  File "/tmp/youtube-dl/youtube_dl/downloader/common.py", line 342, in download
    return self.real_download(filename, info_dict)
  File "/tmp/youtube-dl/youtube_dl/downloader/rtmp.py", line 202, in real_download
    self.report_error('rtmpdump exited with code %d' % retval)
  File "/tmp/youtube-dl/youtube_dl/downloader/common.py", line 155, in report_error
    self.ydl.report_error(*args, **kargs)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 529, in report_error
    self.trouble(error_message, tb)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 491, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())

$ 
````

**Comment by fb39ca4** [comment]
Yep, I was trying from the US. Do their video servers have differing configurations?

**Comment by k1-hedayati** [comment]
Same issue from US

**Comment by keybounce** [comment]
And I'm also getting this problem myself.

There has to be some way to play it -- the browsers are able to negotiate a decryption and play ...

Heck, even if they require us to download the ads in order to download the video, I'm ok with that.

**Comment by Jaegrqualm** [comment]
I don't know If this would help, but using youtube-dl through mpv gives me this.
```
    $ mpv http://www.crunchyroll.com/naruto-shippuden/episode-421-the-sage-of-the-six-paths-677377
Playing: http://www.crunchyroll.com/naruto-shippuden/episode-421-the-sage-of-the-six-paths-677377
     Subs  --sid=1 --slang=en 'enUS ass' (ass) (external)
     Subs  --sid=1 --slang=en 'enUS ass' (ass) (external)
     Subs  --sid=2 --slang=fr 'frFR ass' (ass) (external)
     Subs  --sid=1 --slang=en 'enUS ass' (ass) (external)
     Subs  --sid=2 --slang=fr 'frFR ass' (ass) (external)
     Subs  --sid=3 --slang=it 'itIT ass' (ass) (external)
[ffmpeg] rtmpe: Unknown connect error (unsupported authentication method?)
[ffmpeg] rtmpe: Server error: Connection failed.
Failed to open rtmpe://c2fs.fplive.net/c2?nvb=20150805051824&nva=20150805091924&token=09accaaafbe870f43f62f.>
```

**Comment by Krak-ON** [comment]
The videos are actually stored on Crunchyrolls servers as MP4's.

http://v.lvlt.crcdn.net/c20/s/ve1932381/video.mp4?nvb=20150817042152&nva=20150817082252&token=01d9667bdff43bdb20f07

So what I do is dump the JSON info using -j command.

Then add the necessary info to the URL scheme.

<code>youtube-dl -n -j http://www.crunchyroll.com/baby-steps/episode-20-psychological-warfare-and-self-control-678335

{"display_id": "678335", "upload_date": "20150823", "description": "Eiichiro thought he was successfully ignoring the crowd's roaring cheers for Ide, but his body was reacting to it more than he realized. The pressure of playing in front of a dramatic, Ide-centric crowd is troubling Eiichiro. What can he do to overcome this situation? Eiichiro wrote down the pressure he felt in his notebook and tries to face the details he has been ignoring. Ide is on his game as his crowd cheers on; what will lead to Eiichiro's road to victory?!", "extractor": "Crunchyroll", "requested_subtitles": null, "format": "1080p", "_filename": "Baby Steps 2 Episode 20 \u2013 Psychological Warfare and Self-Control-678335.flv", "uploader": "NHKEnterprise", "format_id": "1080p", "play_path": "mp4:s/ve1932381/video.mp4", "id": "678335", "subtitles": {}, "playlist": null, "thumbnails": [{"url": "http://img1.ak.crunchyroll.com/i/spire3-tmb/cd434d9485dfb012f3caafd32117275d1439592865_full.jpg", "id": "0"}], "title": "Baby Steps 2 Episode 20 \u2013 Psychological Warfare and Self-Control", "url": "rtmpe://c20fs.fplive.net/c20?nvb=20150817042152&nva=20150817082252&token=01d9667bdff43bdb20f07", "extractor_key": "Crunchyroll", "http_headers": {"Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7", "Accept-Language": "en-us,en;q=0.5", "Accept-Encoding": "gzip, deflate", "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/20.0 (Chrome)"}, "playlist_index": null, "ext": "flv", "webpage_url": "http://www.crunchyroll.com/baby-steps/episode-20-psychological-warfare-and-self-control-678335", "formats": [{"http_headers": {"Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7", "Accept-Language": "en-us,en;q=0.5", "Accept-Encoding": "gzip, deflate", "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/20.0 (Chrome)"}, "url": "rtmpe://c20fs.fplive.net/c20?nvb=20150817042150&nva=20150817082250&token=07d52c065ef8034404d89", "play_path": "mp4:s/ve1932375/video.mp4", "format": "360p", "ext": "flv", "format_id": "360p"}, {"http_headers": {"Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7", "Accept-Language": "en-us,en;q=0.5", "Accept-Encoding": "gzip, deflate", "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/20.0 (Chrome)"}, "url": "rtmpe://c20fs.fplive.net/c20?nvb=20150817042150&nva=20150817082250&token=07e0a55a0f5dfb50c6569", "play_path": "mp4:s/ve1932377/video.mp4", "format": "480p", "ext": "flv", "format_id": "480p"}, {"http_headers": {"Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7", "Accept-Language": "en-us,en;q=0.5", "Accept-Encoding": "gzip, deflate", "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/20.0 (Chrome)"}, "url": "rtmpe://c20fs.fplive.net/c20?nvb=20150817042151&nva=20150817082251&token=018b47e5338c4fde32260", "play_path": "mp4:s/ve1932379/video.mp4", "format": "720p", "ext": "flv", "format_id": "720p"}, {"http_headers": {"Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7", "Accept-Language": "en-us,en;q=0.5", "Accept-Encoding": "gzip, deflate", "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/20.0 (Chrome)"}, "url": "rtmpe://c20fs.fplive.net/c20?nvb=20150817042152&nva=20150817082252&token=01d9667bdff43bdb20f07", "play_path": "mp4:s/ve1932381/video.mp4", "format": "1080p", "ext": "flv", "format_id": "1080p"}], "fulltitle": "Baby Steps 2 Episode 20 \u2013 Psychological Warfare and Self-Control", "thumbnail": "http://img1.ak.crunchyroll.com/i/spire3-tmb/cd434d9485dfb012f3caafd32117275d1439592865_full.jpg", "webpage_url_basename": "episode-20-psychological-warfare-and-self-control-678335"}</code>

**Comment by keybounce** [comment]
Thanks for figuring this out! Does this mean it will be automated in Youtube-dl "real soon now"?

**Comment by JamesTheAwesomeDude** [comment]
@Krak-ON That does not fix it. Yes, YouTube-DL can still get _a_ URL, but the authentication is still broken:

````
$ rtmpdump -V -r `youtube-dl http://www.crunchyroll.com/puella-magi-madoka-magica/episode-12-my-very-best-friend-591757 -g`
RTMPDump v2.4
(c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
DEBUG: Parsing...
DEBUG: Parsed protocol: 2
DEBUG: Parsed host    : c9fs.fplive.net
DEBUG: Parsed app     : c9?nvb=20150818121955&nva=20150818162055&token=0962d4bef82a14c5e9b20mp4:s/ve807957
WARNING: You haven't specified an output file (-o filename), using stdout
DEBUG: Protocol : RTMPE
DEBUG: Hostname : c9fs.fplive.net
DEBUG: Port     : 1935
DEBUG: Playpath : mp4:video
DEBUG: tcUrl    : rtmpe://c9fs.fplive.net:1935/c9?nvb=20150818121955&nva=20150818162055&token=0962d4bef82a14c5e9b20mp4:s/ve807957
DEBUG: app      : c9?nvb=20150818121955&nva=20150818162055&token=0962d4bef82a14c5e9b20mp4:s/ve807957
DEBUG: live     : no
DEBUG: timeout  : 30 sec
DEBUG: Setting buffer time to: 36000000ms
Connecting ...
DEBUG: RTMP_Connect1, ... connected, handshaking
DEBUG: HandShake: Client type: 06
DEBUG: HandShake: DH pubkey position: 472
DEBUG: HandShake: Client digest offset: 1383
DEBUG: HandShake: Initial client digest: 
DEBUG: 52 cb cb c8 32 9e d1 33 24 7d fa 3d 04 40 19 b5
DEBUG: 0b be 2f 29 ad 02 83 d8 6a 14 77 28 92 15 63 16
DEBUG: HandShake: Type Answer   : 09
WARNING: HandShake: Type mismatch: client sent 6, server answered 9
DEBUG: HandShake: Server Uptime : 233494510
DEBUG: HandShake: FMS Version   : 4.5.3.1
DEBUG: HandShake: Server DH public key offset: 594
DEBUG: HandShake: Secret key: 
DEBUG: 9f 8e 62 81 10 12 99 9a a3 f8 62 63 d3 25 d6 eb
DEBUG: 59 43 db 90 ea 25 12 9e b0 25 58 d0 92 e8 14 de
DEBUG: a4 2f ab e3 ca e7 0f 89 3a 05 bd 7a 67 94 7a fe
DEBUG: 93 c6 6d 35 50 4f 2a b4 22 b1 0c fe 81 e5 26 b6
DEBUG: 14 34 18 8b 29 b1 7c 43 d1 d2 a6 29 29 f3 77 8d
DEBUG: 14 74 8a c1 f7 b7 a2 6d c8 9a 9f ef b3 c1 f6 4c
DEBUG: e0 01 d3 ca fe b1 9c b4 dc b7 2e 16 9c a5 85 d5
DEBUG: 94 f1 72 35 ea 6c 63 ef ed 33 e4 92 87 4f bc 03
DEBUG: RC4 Out Key: 
DEBUG: 2c 17 f9 b4 f1 9b 4a 31 b6 b2 79 34 fb ad 5a e8
DEBUG: RC4 In Key: 
DEBUG: c6 be bc 69 f2 3a 4e 9a 89 91 47 6a 00 b4 c3 90
DEBUG: HandShake: Calculated digest key from secure key and server digest: 
DEBUG: 18 12 f8 84 ea 4a 03 49 de 73 7d 1d 3b 77 fa 22
DEBUG: 46 3f d8 23 29 58 9f 5b 07 e0 5e c8 1c 22 22 06
DEBUG: HandShake: Client signature calculated:
DEBUG: ba 1e ff 9f f4 ee e1 26 21 1b 24 42 e6 99 5b d0
DEBUG: 5c 66 9f af 89 ac a6 e0 26 12 1f 5a b5 2c 63 5c
DEBUG: HandShake: Server sent signature:
DEBUG: 9a 18 85 83 f1 9a 3b 07 c9 ba 74 56 d7 5a 54 45
DEBUG: 82 05 41 f2 08 41 11 1a 28 e1 a6 63 e2 5d ea b0
DEBUG: HandShake: Digest key: 
DEBUG: 81 80 f9 33 86 04 93 51 04 30 65 e1 8c 76 c7 79
DEBUG: 0b 9e 4d db b6 04 10 d5 0d 29 8d 96 a0 08 ec 21
DEBUG: HandShake: Signature calculated:
DEBUG: 9a 18 85 83 f1 9a 3b 07 c9 ba 74 56 d7 5a 54 45
DEBUG: 82 05 41 f2 08 41 11 1a 28 e1 a6 63 e2 5d ea b0
DEBUG: HandShake: Genuine Adobe Flash Media Server
DEBUG: HandShake: Handshaking finished....
DEBUG: RTMP_Connect1, handshaked
DEBUG: Invoking connect
INFO: Connected...
DEBUG: HandleChangeChunkSize, received: chunk size change to 128
DEBUG: RTMP_ClientPacket, received: invoke 165 bytes
DEBUG: (object begin)
DEBUG: Property: NULL
DEBUG: (object begin)
DEBUG: Property: <Name:              level, STRING:	error>
DEBUG: Property: <Name:               code, STRING:	NetConnection.Connect.Rejected>
DEBUG: Property: <Name:        description, STRING:	Connection failed.>
DEBUG: Property: <Name:        description, STRING:	[ Server.Reject ] : Connection failed.>
DEBUG: (object end)
DEBUG: (object end)
DEBUG: HandleInvoke, server invoking <_error>
ERROR: rtmp server sent error
DEBUG: RTMP_ClientPacket, received: invoke 18 bytes
DEBUG: (object begin)
DEBUG: Property: NULL
DEBUG: (object end)
DEBUG: HandleInvoke, server invoking <close>
ERROR: rtmp server requested close
DEBUG: Closing connection.

$ 
````

Running it with the -g flag and then running rtmpdump separately instead of just letting YTDL call rtmpdump does not fix anything.

**Comment by JamesTheAwesomeDude** [comment]
It looks like, according to [this guy](http://stream-recorder.com/forum/showpost.php?p=26212&postcount=2), Crunchyroll is simply using a form of security too great to be cracked by rtmpdump.

We'll need to wait for the people behind rtmpdump to break the v9 handshake system before we can download from Crunchyroll again.

**Comment by JamesTheAwesomeDude** [comment]
I can confirm that what @Krak-ON tried works.

Take the URL `rtmpe://c$Nfs.fplive.net/c$N?nvb=$NVB&nva=$NVA&token=$TOKEN` and the play_path `mp4:s/$VE/video.mp4` and change it to `http://v.lvlt.crcdn.net/c$N/s/$VE/video.mp4\?nvb=$NVB&nva=$NVA&token=$TOKEN` and you'll have something that can be fetched with just wget.

**Comment by JamesTheAwesomeDude** [comment]
This didn't fix it.

````
$ git fetch
Enter passphrase for key '/home/james/.ssh/id_ecdsa': 
remote: Counting objects: 23, done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 23 (delta 11), reused 1 (delta 1), pack-reused 0
Unpacking objects: 100% (23/23), done.
From github.com:rg3/youtube-dl
   3b9b32f..a01da8b  master     -> origin/master
$ python3 -m youtube_dl -v http://www.crunchyroll.com/puella-magi-madoka-magica/episode-12-my-very-best-friend-591757
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-v', 'http://www.crunchyroll.com/puella-magi-madoka-magica/episode-12-my-very-best-friend-591757']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.08.16.1
[debug] Git HEAD: 3b9b32f
[debug] Python version 3.4.0 - Linux-3.16.0-34-generic-x86_64-with-LinuxMint-17.1-rebecca
[debug] exe versions: ffmpeg N-74369-g55a07cf, ffprobe N-74369-g55a07cf, rtmpdump 2.4
[debug] Proxy map: {}
[Crunchyroll] 591757: Downloading webpage
[Crunchyroll] 591757: Downloading media info
[Crunchyroll] 591757: Downloading media info for 360p
[Crunchyroll] 591757: Downloading media info for 480p
[Crunchyroll] 591757: Downloading media info for 720p
[Crunchyroll] 591757: Downloading media info for 720p
[Crunchyroll] 591757: Downloading media info for 1080p
[Crunchyroll] 591757: Downloading media info for 1080p
[debug] Invoking downloader on 'rtmpe://c9fs.fplive.net/c9?nvb=20150818170752&nva=20150818210852&token=0ae16a4f53aa749960b92'
[download] Destination: Puella Magi Madoka Magica Episode 12 – My Very Best Friend-591757.flv
[debug] rtmpdump command line: rtmpdump --verbose -r 'rtmpe://c9fs.fplive.net/c9?nvb=20150818170752&nva=20150818210852&token=0ae16a4f53aa749960b92' -o 'Puella Magi Madoka Magica Episode 12 – My Very Best Friend-591757.flv.part' --playpath mp4:s/ve807957/video.mp4 --resume --skip 1
[rtmpdump] RTMPDump v2.4
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 2
[rtmpdump] DEBUG: Parsed host    : c9fs.fplive.net
[rtmpdump] DEBUG: Parsed app     : c9?nvb=20150818170752&nva=20150818210852&token=0ae16a4f53aa749960b92
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMPE
[rtmpdump] DEBUG: Hostname : c9fs.fplive.net
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : mp4:s/ve807957/video.mp4
[rtmpdump] DEBUG: tcUrl    : rtmpe://c9fs.fplive.net:1935/c9?nvb=20150818170752&nva=20150818210852&token=0ae16a4f53aa749960b92
[rtmpdump] DEBUG: app      : c9?nvb=20150818170752&nva=20150818210852&token=0ae16a4f53aa749960b92
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.
[rtmpdump] 0 bytes
[rtmpdump] RTMPDump v2.4
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 2
[rtmpdump] DEBUG: Parsed host    : c9fs.fplive.net
[rtmpdump] DEBUG: Parsed app     : c9?nvb=20150818170752&nva=20150818210852&token=0ae16a4f53aa749960b92
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMPE
[rtmpdump] DEBUG: Hostname : c9fs.fplive.net
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : mp4:s/ve807957/video.mp4
[rtmpdump] DEBUG: tcUrl    : rtmpe://c9fs.fplive.net:1935/c9?nvb=20150818170752&nva=20150818210852&token=0ae16a4f53aa749960b92
[rtmpdump] DEBUG: app      : c9?nvb=20150818170752&nva=20150818210852&token=0ae16a4f53aa749960b92
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.


ERROR: rtmpdump exited with code 1
  File "/usr/lib/python3.4/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.4/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/youtube-dl/youtube_dl/__main__.py", line 19, in <module>
    youtube_dl.main()
  File "/tmp/youtube-dl/youtube_dl/__init__.py", line 410, in main
    _real_main(argv)
  File "/tmp/youtube-dl/youtube_dl/__init__.py", line 400, in _real_main
    retcode = ydl.download(all_urls)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 1653, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 666, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 712, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 1323, in process_video_result
    self.process_info(new_info)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 1585, in process_info
    success = dl(filename, info_dict)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 1527, in dl
    return fd.download(name, info)
  File "/tmp/youtube-dl/youtube_dl/downloader/common.py", line 342, in download
    return self.real_download(filename, info_dict)
  File "/tmp/youtube-dl/youtube_dl/downloader/rtmp.py", line 202, in real_download
    self.report_error('rtmpdump exited with code %d' % retval)
  File "/tmp/youtube-dl/youtube_dl/downloader/common.py", line 155, in report_error
    self.ydl.report_error(*args, **kargs)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 530, in report_error
    self.trouble(error_message, tb)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 492, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())

$ 
````

**Comment by dstftw** [comment]
Wrong.
```
[debug] Git HEAD: 3b9b32f
```

**Comment by JamesTheAwesomeDude** [comment]
Ah, my mistake. After deleting the folder and re-fetching it worked! Thanks for this fix!

How do I make it accept the git updates without nuking the whole git dir, by the way? I tried `find -name "__pycache__" -exec rm -rfv {} \;` but that didn't do it.

**Comment by dstftw** [comment]
`git pull`.

**Comment by keybounce** [comment]
```
keybounceMBP:Play1710 michael$ youtube-dl -U
youtube-dl is up-to-date (2015.08.16.1)
```
About how long until this pushes for ordinary users?

**Comment by dstftw** [comment]
Up to a week.

**Comment by keybounce** [comment]
OK.

```
keybounceMBP:Programming michael$ git clone https://github.com/rg3/youtube-dl.git
Cloning into 'youtube-dl'...
remote: Counting objects: 51062, done.
remote: Compressing objects: 100% (119/119), done.
remote: Total 51062 (delta 71), reused 0 (delta 0), pack-reused 50941
Receiving objects: 100% (51062/51062), 32.64 MiB | 1.54 MiB/s, done.
Resolving deltas: 100% (35859/35859), done.
Checking connectivity... done.
keybounceMBP:Programming michael$ cd youtube-dl/
keybounceMBP:youtube-dl michael$ ls
total 100
  4 AUTHORS                      56 README.md                     4 setup.py
 12 CONTRIBUTING.md               0 bin/                          0 test/
  4 LICENSE                       0 devscripts/                   4 tox.ini
  4 MANIFEST.in                   0 docs/                         4 youtube-dl.plugin.zsh
  4 Makefile                      4 setup.cfg                     0 youtube_dl/
keybounceMBP:youtube-dl michael$ make
zip --quiet youtube-dl youtube_dl/*.py youtube_dl/*/*.py
zip --quiet --junk-paths youtube-dl youtube_dl/__main__.py
echo '#!/usr/bin/env python' > youtube-dl
cat youtube-dl.zip >> youtube-dl
rm youtube-dl.zip
chmod a+x youtube-dl
pandoc -f markdown -t plain README.md -o README.txt
make: pandoc: No such file or directory
make: *** [README.txt] Error 1
keybounceMBP:youtube-dl michael$ 
```

So:
1. It's just a zip with the interpreter header? OK.
2. Do I need to worry about the error (probably not, it's a read me), or any skipped steps (unknown what wasn't done)?

**Comment by dstftw** [comment]
https://github.com/rg3/youtube-dl#developer-instructions

**Comment by Krak-ON** [comment]
@keybounce Use <code>make youtube-dl</code> to not get an error.

**Comment by keybounce** [comment]
So, attempting to use the now current regular version (2015.08.28) is still failing.

youtube-dl -f 'bestvideo[height<=480][ext=mp4]' --sub-lang enUS --write-sub --recode-video mp4 http://www.crunchyroll.com/is-it-wrong-to-try-to-pick-up-girls-in-a-dungeon
[crunchyroll:playlist] is-it-wrong-to-try-to-pick-up-girls-in-a-dungeon: Downloading webpage
[download] Downloading playlist: Is It Wrong to Try to Pick Up Girls in a Dungeon?
[crunchyroll:playlist] playlist Is It Wrong to Try to Pick Up Girls in a Dungeon?: Collected 13 video ids (downloading 13 of them)
[download] Downloading video 1 of 13
[Crunchyroll] 678135: Downloading webpage
[Crunchyroll] 678135: Downloading media info
[Crunchyroll] 678135: Downloading media info for 360p
[Crunchyroll] 678135: Checking 360p URL
[Crunchyroll] 678135: Downloading media info for 480p
[Crunchyroll] 678135: Checking 480p URL
[Crunchyroll] 678135: Downloading media info for 720p
[Crunchyroll] 678135: Checking 720p URL
[Crunchyroll] 678135: Downloading media info for 720p
[Crunchyroll] 678135: Checking 720p URL
[Crunchyroll] 678135: Downloading media info for 1080p
[Crunchyroll] 678135: Checking 1080p URL
[Crunchyroll] 678135: Downloading media info for 1080p
[Crunchyroll] 678135: Checking 1080p URL
[Crunchyroll] 678135: Downloading subtitles for English (US)
[Crunchyroll] 678135: Downloading subtitles for Español
[Crunchyroll] 678135: Downloading subtitles for Português (Brasil)
[Crunchyroll] 678135: Downloading subtitles for العربية
ERROR: requested format not available

What am I doing wrong?

**Comment by yan12125** [comment]
```
ERROR: requested format not available
```
Try to update the format filter ```'bestvideo[height<=480][ext=mp4]'```

**Comment by Krak-ON** [comment]
Try <code>youtube-dl --write-sub --sub-format ass --sub-lang enUS --embed-subs --recode-video mkv</code>

**Comment by Krak-ON** [comment]
@keybounce The videos are already in a MP4.

**Comment by keybounce** [comment]
Ok; what's the proper format string to use?

My goal is to have a single wrapper / macro / script that can call youtube-dl, and fetch a 480p mp4, or as close to 480 as I can. So far, this has worked with flv's and mp4's, and is probably the 3rd or 4th iteration of improvement.

Normally, if it's flv, it converts, and if it's mp4, just leaves it as-is. So what else do I need?

```
keybounceMBP:Finite-Fluids michael$ cat ~/bin/youtube-480 
#!/bin/bash
set -x
# Usage: youtube-480 url
## youtube-dl -f "bestvideo[height<=480][ext=mp4]+bestaudio[ext=m4a]/[height<=?480]" --sub-lang en --write-sub --recode-video mp4 "$@"
youtube-dl -f "bestvideo[height<=480][ext=mp4]" --list-sub "$@" | grep -m 1 '^[eE][nN]' | awk '{print $1}' > /tmp/yt-sub.$$
youtube-dl -f "bestvideo[height<=480][ext=mp4]" --sub-lang $(cat /tmp/yt-sub.$$) --write-sub --recode-video mp4 "$@"
```

**Comment by yan12125** [comment]
If you specify ```--recode-video mp4```, ```[ext=mp4]``` in the format filter may be unnecessary. All downloaded files, either in flv or mp4, will be transformed into mp4.
Also, please open a new issue the next time. Your problem is different from the original one.

**Comment by maexono** [comment]
In Germany I still have the original issue. Though, interestingly, the rtmp-URLs look completely different:
```
>youtube-dl -n -v http://www.crunchyroll.com/ushio-and-tora/episode-1-the-fate-that-brings-ushio-and-tora-together-682425
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-n', '-v', 'http://www.crunchyroll.com/ushio-and-tora/episode-1-the-fate-that-brings-ushio-and-tora-together-682425']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2015.08.28
[debug] Python version 3.4.3 - Windows-8-6.2.9200
[debug] exe versions: ffmpeg 2.5.2, ffprobe 2.5.2, rtmpdump 2.4
[debug] Proxy map: {}
[Crunchyroll] Logging in
[Crunchyroll] 682425: Downloading webpage
[Crunchyroll] 682425: Downloading media info
[Crunchyroll] 682425: Downloading media info for 360p
[Crunchyroll] 682425: Downloading media info for 480p
[Crunchyroll] 682425: Downloading media info for 720p
[Crunchyroll] 682425: Downloading media info for 1080p
[debug] Invoking downloader on 'rtmpe://cp150757.edgefcs.net/ondemand/?auth=daEdxbvaddwdad6braOaMdGc3cTdZcQdHb.-bv5QDs-dHa-oBEwotJDxxC&aifp=0009&slist=c18/s/ve1882337/video.mp4'
[download] Destination: Ushio and Tora Episode 1 – The Fate that Brings Ushio And Tora Together-682425.flv
[debug] rtmpdump command line: rtmpdump --verbose -r 'rtmpe://cp150757.edgefcs.net/ondemand/?auth=daEdxbvaddwdad6braOaMdGc3cTdZcQdHb.-bv5QDs-dHa-oBEwotJDxxC&aifp=0009&slist=c18/s/ve1882337/video.mp4' -o 'Ushio and Tora Episode 1 – The Fate that Brings Ushio And Tora Together-682425.flv.part' --playpath mp4:c18/s/ve1882337/video.mp4 --resume --skip 1
[rtmpdump] RTMPDump v2.4
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 2
[rtmpdump] DEBUG: Parsed host    : cp150757.edgefcs.net
[rtmpdump] DEBUG: Parsed app     : ondemand/?auth=daEdxbvaddwdad6braOaMdGc3cTdZcQdHb.-bv5QDs-dHa-oBEwotJDxxC&aifp=0009&slist=c18/s/ve1882337/video.mp4
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMPE
[rtmpdump] DEBUG: Hostname : cp150757.edgefcs.net
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : mp4:c18/s/ve1882337/video.mp4
[rtmpdump] DEBUG: tcUrl    : rtmpe://cp150757.edgefcs.net:1935/ondemand/?auth=daEdxbvaddwdad6braOaMdGc3cTdZcQdHb.-bv5QDs-dHa-oBEwotJDxxC&aifp=0009&slist=c18/s/ve1882337/video.mp4
[rtmpdump] DEBUG: app      : ondemand/?auth=daEdxbvaddwdad6braOaMdGc3cTdZcQdHb.-bv5QDs-dHa-oBEwotJDxxC&aifp=0009&slist=c18/s/ve1882337/video.mp4
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.
[rtmpdump] 0 bytes
[rtmpdump] RTMPDump v2.4
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 2
[rtmpdump] DEBUG: Parsed host    : cp150757.edgefcs.net
[rtmpdump] DEBUG: Parsed app     : ondemand/?auth=daEdxbvaddwdad6braOaMdGc3cTdZcQdHb.-bv5QDs-dHa-oBEwotJDxxC&aifp=0009&slist=c18/s/ve1882337/video.mp4
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMPE
[rtmpdump] DEBUG: Hostname : cp150757.edgefcs.net
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : mp4:c18/s/ve1882337/video.mp4
[rtmpdump] DEBUG: tcUrl    : rtmpe://cp150757.edgefcs.net:1935/ondemand/?auth=daEdxbvaddwdad6braOaMdGc3cTdZcQdHb.-bv5QDs-dHa-oBEwotJDxxC&aifp=0009&slist=c18/s/ve1882337/video.mp4
[rtmpdump] DEBUG: app      : ondemand/?auth=daEdxbvaddwdad6braOaMdGc3cTdZcQdHb.-bv5QDs-dHa-oBEwotJDxxC&aifp=0009&slist=c18/s/ve1882337/video.mp4
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.


ERROR: rtmpdump exited with code 1
  File "C:\tools\python3\lib\runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\tools\python3\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\tools\python3\Scripts\youtube-dl.exe\__main__.py", line 9, in <module>
    sys.exit(main())
  File "C:\tools\python3\lib\site-packages\youtube_dl\__init__.py", line 410, in main
    _real_main(argv)
  File "C:\tools\python3\lib\site-packages\youtube_dl\__init__.py", line 400, in _real_main
    retcode = ydl.download(all_urls)
  File "C:\tools\python3\lib\site-packages\youtube_dl\YoutubeDL.py", line 1653, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "C:\tools\python3\lib\site-packages\youtube_dl\YoutubeDL.py", line 666, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "C:\tools\python3\lib\site-packages\youtube_dl\YoutubeDL.py", line 712, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "C:\tools\python3\lib\site-packages\youtube_dl\YoutubeDL.py", line 1323, in process_video_result
    self.process_info(new_info)
  File "C:\tools\python3\lib\site-packages\youtube_dl\YoutubeDL.py", line 1585, in process_info
    success = dl(filename, info_dict)
  File "C:\tools\python3\lib\site-packages\youtube_dl\YoutubeDL.py", line 1527, in dl
    return fd.download(name, info)
  File "C:\tools\python3\lib\site-packages\youtube_dl\downloader\common.py", line 342, in download
    return self.real_download(filename, info_dict)
  File "C:\tools\python3\lib\site-packages\youtube_dl\downloader\rtmp.py", line 202, in real_download
    self.report_error('rtmpdump exited with code %d' % retval)
  File "C:\tools\python3\lib\site-packages\youtube_dl\downloader\common.py", line 155, in report_error
    self.ydl.report_error(*args, **kargs)
  File "C:\tools\python3\lib\site-packages\youtube_dl\YoutubeDL.py", line 530, in report_error
    self.trouble(error_message, tb)
  File "C:\tools\python3\lib\site-packages\youtube_dl\YoutubeDL.py", line 492, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())
```
